### PR TITLE
[Copy] Update skill library table

### LINKED ISSUE #8091: ♻️ Make minor modifications to the candidate-facing skill table
image

## ✨ Summary

[Visit the Figma file](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=108-39250&mode=design&t=JfScQNdCl3fNhF2Z-4) • Chat with @substrae for design questions

We noticed a few small state-related things, as well as would like to make a few small copy enhancements. First and foremost, when performing a search and no results are found, the table is missing a null state. We've also updated the "add" button text and default empty null state text.

```[tasklist]
### ✅ Requirements
- [ ] Update the create button's label to "Add a new skill"
- [ ] Update the default empty state's text to match
- [ ] Add a message for when no search results exist
``` 

## 🔍 Details

<details>
  <summary>📸 Default null state</summary>
  
</details> 

<details>
  <summary>📸 No search results</summary>
  
</details>

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8091 

## 👋 Introduction

Updates the copy for the skill library table.

## 🕵️ Details

- Add button renamed "Add a new skill"
- Null message updated
- Added new "no search results" message

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Navigate to your skill library `/applicant/profile-and-applications/skills`
3. Confirm add button reads "Add a new skill"
4. Enter a search term that results in no rows being returned
5. Confirm new "no search results" message
6. Delete all skills in a table
7. Confirm the new null message appears

## 📸 Screenshot

Screenshot 2023-10-10 101411
Screenshot 2023-10-10 101415

### CODE REVIEW

File: apps/web/src/pages/Skills/components/SkillLibraryTable.tsx [hunk 1]
```diff
@@ -181,6 +181,33 @@ const SkillLibraryTable = ({
         total: data.length,
         pageSizes: [10, 20, 50],
       }}
+      nullMessage={{
+        title: intl.formatMessage({
+          defaultMessage: "There are'nt any skills here yet.",
```
Comment (vd1992) [inline apps/web/src/pages/Skills/components/SkillLibraryTable.tsx:186]: Typo

**Comment by vd1992** [comment]
The screenshots in the issue contain placeholder text and an icon for the search bar, are those to be added sometime?
The search bar doesn't have them currently but has the spacing for an icon.

image

**Comment by esizer** [comment]
> The screenshots in the issue contain placeholder text and an icon for the search bar, are those to be added sometime?
> The search bar doesn't have them currently but has the spacing for an icon.

I'm not sure. It was not part of the AC so I assume it was just a change that happened earlier and was no reflected in the design. I know we did [remove the icon in the table polish PR](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7931/commits/7a52813007c09863dfa7ab2816ca598030710cbe).

**Comment by vd1992** [comment]
> > The screenshots in the issue contain placeholder text and an icon for the search bar, are those to be added sometime?
> > The search bar doesn't have them currently but has the spacing for an icon.
> 
> I'm not sure. It was not part of the AC so I assume it was just a change that happened earlier and was no reflected in the design. I know we did [remove the icon in the table polish PR](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7931/commits/7a52813007c09863dfa7ab2816ca598030710cbe).

As long as we're aware of it and it appears intended, then fine by me

**Review by vd1992 — APPROVED** [review]

**Pull request merged**